### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mhamidjamil**
Orange pi can now communicate with TTGO TCall module.
it ill reply if ttgo tcall ask for time

### CODE REVIEW

File: serial_communication/web/app.py [hunk 1]
```diff
@@ -1,20 +1,63 @@
...
+    # Get a list of available serial ports with default baud rate
+    available_ports = get_serial_ports()
+
+    # Baud rates to be displayed in the dropdown
+    baud_rates = [9600, 115200, 230400, 460800, 921600]
+
+    return render_template('index.html', default_port=default_serial_port, available_ports=available_ports, baud_rates=baud_rates)
 
 
+def read_serial_data(data):
```
Comment (wasey-rao) [inline serial_communication/web/app.py:43]: Please write a separate file for input management.

File: serial_communication/web/app.py [hunk 2]
```diff
@@ -24,5 +67,41 @@ def send_serial():
     return {'status': 'success'}
 
 
+def fetch_current_time_online():
```
Comment (wasey-rao) [inline serial_communication/web/app.py:70]: Don't use fetch_time....py code here, do it in separate file just call it form this code.

File: serial_communication/web/templates/index.html [hunk 3]
```diff
@@ -42,12 +49,14 @@ <h1>Serial Interface</h1>
             fetch('/read_serial')
                 .then(response => response.json())
                 .then(data => {
-                    // Append the new data to the existing content
+                    // Append the new data to the existing content with different color for received data
                     const serialData = document.getElementById('serial-data');
-                    serialData.innerText += data.data + '\n';
```
Comment (wasey-rao) [inline serial_communication/web/templates/index.html:47]: I don't think you input data is visible in web terminal, fix it.

**Review by wasey-rao — CHANGES_REQUESTED** [review]
Please resolve comments then ping me for approval.

File: serial_communication/namaz/time_fetcher.py
Comment (mhamidjamil) [inline serial_communication/namaz/time_fetcher.py:1]: if it is Isha time it don't show the coming Fajir time.

**Review by wasey-rao — APPROVED** [review]

**Comment by wasey-rao** [comment]
I'm merging this code just because of Namaz time file, other wise kindly resolve comments before applying any changes in PR or you better create new PR for new feature.

**Pull request merged**